tests/de/gurkenlabs/litiengine/physics/CollisionResolvingTests.java

```diff
@@ -29,195 +32,155 @@ public void clear() {
-    assertEquals(10, ent.getY(), EPSILON);
-
-    // move back 10 px to the left
-    Game.physics().move(ent, 270, 10);
+    // act
+    Game.physics().move(ent, angle, distance);
 
-    assertEquals(10, ent.getX(), EPSILON);
+    // assert
+    // block movement for remaining 5px
+    assertEquals(targetX, ent.getX(), EPSILON);
+    assertEquals(targetY, ent.getY(), EPSILON);
+  }
 
-    // now "slide" along the rectangle to the bottom right
-    Game.physics().move(ent, 45, 14.14213562373095);
+  @ParameterizedTest(name = "testCollidingMoveSlide: {0}")
+  @MethodSource("supplyCollidingMoveSlideParameters")
+  public void testCollidingMoveSlide(String direction, CollisionBox collisionBox, int angle, int targetX, int targetY) {
+    // arrange
+    Creature ent = getNewCreature(); // pos: (10,10), w/h: 10/10
+    Game.physics().add(ent);
 
-    assertEquals(20, ent.getX(), EPSILON);
-    assertEquals(10, ent.getY(), EPSILON);
+    // large rectangle right next to the entity
+    Game.physics().add(collisionBox);
+    Game.physics().update();
 
-    // now "slide" along the rectangle to the bottom left
-    Game.physics().move(ent, 315, 14.14213562373095);
+    // act
+    // "slide" along the rectangle in direction
+    Game.physics().move(ent, angle, MOVE_10X10Y_DISTANCE);
 
-    assertEquals(10, ent.getX(), EPSILON);
-    assertEquals(10, ent.getY(), EPSILON);
+    // assert
+    assertEquals(targetX, ent.getX(), EPSILON);
+    assertEquals(targetY, ent.getY(), EPSILON);
   }
 
   @Test
-  public void testCollidingVerticalMovement() {
-    Creature ent = getNewCreature();
-
+  public void testCollidingMoveSlideDown() {
+    // arrange
+    Creature ent = getNewCreature(); // pos: (10,10), w/h: 10/10
     Game.physics().add(ent);
 
-    // large rectangle at the right of the entity
-    Game.physics().add(new CollisionBox(20, 0, 10, 100));
-
-    // move 10 px down
+    // large rectangle to the left of the entity
+    Game.physics().add(new CollisionBox(0, 0, 10, 100));
     Game.physics().update();
-    Game.physics().move(ent, 0, 10);
-
-    assertEquals(10, ent.getX(), EPSILON);
-    assertEquals(20, ent.getY(), EPSILON);
 
-    // now "slide" along the rectangle to the bottom right
-
-    Game.physics().move(ent, 45, 14.14213562373095);
-
-    assertEquals(10, ent.getX(), EPSILON);
-    assertEquals(30, ent.getY(), EPSILON);
-
-    // now "slide" along the rectangle to the top left
-    Game.physics().move(ent, 135, 14.14213562373095);
+    // act
+    // "slide" along the rectangle to the bottom left
+    // TODO: target collision box wrongfully detects an intersection with blocking collision box due to floating point precision miscalculation
+    //  within method GeometricUtilities.intersects - intersection is marginally positive in case of left movement, but marginally negative in right movement
+    // Game.physics().move(ent, 315, MOVE_10X10Y_DISTANCE);
+    Game.physics().move(ent, 315, MOVE_10X10Y_DISTANCE-(1e-14));
```

Comment:
@nightm4re94 @Gamebuster19901 During development we found this issue with floating point precision during calculating collisions and intersections.
Basically the test will fail if I provide the constant as it is, but it will pass if I subtract a marginal value. The issue is that due to the floating point precision it miscalculates an intersection to exist where there should not be any...
How should we handle this problem?

Reply:
is 1e-14 the order of magnitude of the difference in this example? 
I think it's odd that a number as small as `0.00000000000001` has an effect on the floating number equality assertion, as we've defined the tolerance threshold `EPSILON = 1e-6 = 0.000001`.

Reply:
Exactly my point. However, the problem is not the threshold of `EPSILON` (since it is only relevant for the assertion), but lies in the underlying computations within `GeometricUtilities.intersects()`. The small delta seems to trigger a floating point change that results in finding an intersection between the two `CollisionBox`es when they really just touch instead of intersect

Reply:
ah, I see! Thanks for clarifying!

Reply:
to come back to the original question: how should we proceed with this?
Do you guys work with the Github Issues as backlog (i.e., should we open an issue and ignore the problem for now in this PR), or should we include a potentially larger refactoring in this PR?

Reply:
I think this would best be tracked separately - feel free to open an issue.

Reply:
created an issue [here](https://github.com/gurkenlabs/litiengine/issues/400)